pkger/service.go

```diff
@@ -654,75 +659,74 @@ func (s *Service) DryRun(ctx context.Context, orgID, userID influxdb.ID, pkg *Pk
 	if len(opt.EnvRefs) > 0 {
 		err := pkg.applyEnvRefs(opt.EnvRefs)
 		if err != nil && !IsParseErr(err) {
-			return Summary{}, Diff{}, internalErr(err)
+			return Summary{}, Diff{}, nil, internalErr(err)
 		}
 		parseErr = err
 	}
 
+	state := newStateCoordinator(pkg)
+
 	if opt.StackID > 0 {
-		if err := s.addStackPkgState(ctx, opt.StackID, pkg); err != nil {
-			return Summary{}, Diff{}, internalErr(err)
+		if err := s.addStackState(ctx, opt.StackID, pkg, state); err != nil {
+			return Summary{}, Diff{}, nil, internalErr(err)
 		}
 	}
 
 	if err := s.dryRunSecrets(ctx, orgID, pkg); err != nil {
-		return Summary{}, Diff{}, err
+		return Summary{}, Diff{}, nil, err
 	}
 
-	diff := Diff{
-		Buckets:    s.dryRunBuckets(ctx, orgID, pkg),
-		Checks:     s.dryRunChecks(ctx, orgID, pkg),
-		Dashboards: s.dryRunDashboards(pkg),
-		Labels:     s.dryRunLabels(ctx, orgID, pkg),
-		Tasks:      s.dryRunTasks(pkg),
-		Telegrafs:  s.dryRunTelegraf(pkg),
-		Variables:  s.dryRunVariables(ctx, orgID, pkg),
-	}
+	s.dryRunBuckets(ctx, orgID, state.mBuckets)
```

Comment:
state is added to the stateCoordinator here. TLDR; it looks for existing resources, either by ID or name, then adds the state to the valid state type that maps state from the platform to the pkg resource.